src/main/kotlin/org/dif/peerdid/PeerDIDUtils.kt

```diff
@@ -76,37 +80,55 @@ private fun isInEncodingTypes(encodingAlgorithm: Char): Boolean {
     return EncodingType.values().any { type -> type.type == encodingAlgorithm }
 }
 
-private fun decodeService(encodedService: JSON, peerDID: PeerDID): Map<String, String> {
+private fun decodeService(encodedService: JSON, peerDID: PeerDID): List<Map<String, String>> {
     val decodedService = Base64.decodeBase64(encodedService).decodeToString()
     val gson = GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create()
-    val serviceMap = gson.fromJson(decodedService, HashMap::class.java)
-    val serviceType = serviceMap.remove("t").toString().replace("dm", "didcommmessaging")
-
-    return mapOf(
-        "id" to peerDID.plus('#').plus(serviceType),
-        "type" to serviceType,
-        "serviceEndpoint" to serviceMap.remove("s").toString(),
-        "routingKeys" to serviceMap.remove("r").toString()
-    )
+    val serviceMapList = try {
+        gson.fromJson(decodedService, List::class.java) as List<LinkedTreeMap<String, String>>
+    } catch (e: JsonSyntaxException) {
+        listOf(gson.fromJson(decodedService, HashMap::class.java))
+    }
+    val finalServiceMapList = mutableListOf<Map<String, String>>()
+    for (serviceMap in serviceMapList) {
+        val serviceType = serviceMap.remove("t").toString().replace("dm", "didcommmessaging")
+
+        val service = mapOf(
+            "id" to peerDID.plus('#').plus(serviceType),
```

Comment:
What if we have multiple services (array)? They all will have the same id? If so, that must be fixed, id must be unique.
Probably we need to check it for peer-did python as well.